src/render/draw_custom.js

```diff
@@ -7,15 +7,19 @@ import StencilMode from '../gl/stencil_mode.js';
 import {warnOnce} from '../util/util.js';
 
 import type Painter from './painter.js';
+import type {OverscaledTileID} from '../source/tile_id.js';
 import type SourceCache from '../source/source_cache.js';
 import type CustomStyleLayer from '../style/style_layer/custom_style_layer.js';
+import MercatorCoordinate from '../geo/mercator_coordinate.js';
+import assert from 'assert';
 
-function drawCustom(painter: Painter, sourceCache: SourceCache, layer: CustomStyleLayer) {
+function drawCustom(painter: Painter, sourceCache: SourceCache, layer: CustomStyleLayer, coords: Array<OverscaledTileID>) {
 
     const context = painter.context;
     const implementation = layer.implementation;
 
-    if (painter.transform.projection.unsupportedLayers && painter.transform.projection.unsupportedLayers.includes("custom")) {
+    if (painter.transform.projection.unsupportedLayers && painter.transform.projection.unsupportedLayers.includes("custom") &&
+        !(painter.terrain && (painter.terrain.renderingToTexture || painter.renderPass === 'offscreen') && layer.isLayerDraped())) {
         warnOnce('Custom layers are not yet supported with non-mercator projections. Use mercator to enable custom layers.');
         return;
     }
```

Comment:
Should this error message be updated to note that globe support? I also think the second condition should only early-return, not output an error message:

```suggestion
     if (painter.terrain && !painter.terrain.renderingToTexture && painter.renderPass !== 'offscreen' && layer.isLayerDraped()) {
         // When terrain is enabled, don't process custom draped layers unless we're in the offscreen pass
         return;
    }
    if (painter.transform.projection.unsupportedLayers && painter.transform.projection.unsupportedLayers.includes("custom")) {
        warnOnce('Custom layers are not yet supported with this projection. Use mercator or globe to enable usage of custom layers.');
        return;
    }
```

Reply:
Thanks. Updated comment. Custom layer draping follows the order of other layers and draping happens within the translucent pass.